### ISSUE TITLE
Error messages, if image too big

### ISSUE THREAD

**Issue by ddeimeke** [s9y/Serendipity#684]
It would be nice to get an error message if a user tries to upload to upload an image which is larger than the allowed size (php setting) to the media database.

I find the error message only in the server error log:

`[Thu Mar 26 10:57:54.759775 2020] [php7:warn] [pid 25560] [client 2001:171b:c9bb:cdc0:41b7:54ef:86d8:16eb:*] PHP Warning:  POST Content-Length of 18625471 bytes exceeds the limit of 10485760 bytes in Unknown on line 0, referer: https://training.yawnrz.com/serendipity_admin.php?serendipity[adminModule]=media&serendipity[adminAction]=addSelect`

**Comment by onli** [comment]
I remember seeing a valid error message, but this might depend in the php configuration, or maybe I'm misremembering. In any case I agree, it would be good to properly report the filesize too big error, if we can detect it.

**Comment by th-h** [comment]
Yep. Is there an easy  way to detect that? I think at least the upload operation has to fail?

**Comment by garvinhicking** [comment]
You can only detect the upload limit prior to that by using HTML5 AJAX/XHR based upload tools, not natively within PHP itself. PHP would only be able to detect a larger file than the POST size once it died.

By using HTML5 file upload, the client can propagate the file size, allowing for client-based feedback when a filesize exceeds a configured limit.

**Comment by garvinhicking** [comment]
(HTML5-based image upload provides nice other feature like thumbnail preview, uploading multiple images at once, chunked upload, drag+drop upload. Downside is, it's a bit complex, needs some specific browser workarounds [usually reuqires a library, which then creates licensing issues])

**Comment by onli** [comment]
Well, we do have a js upload when the resizing option is one. The code is a but spaghetti, but adding the check there would be possible. But do we really get no useable information from php when the upload fails?
